packages/tspec/src/generator/openapiGenerator.ts

```diff
@@ -108,8 +108,7 @@ export const getOpenapiPaths = (
     const queryParams = getObjectPropertyByPath(spec, 'query', openapiSchemas) as any;
     const headerParams = getObjectPropertyByPath(spec, 'header', openapiSchemas) as any;
     const cookieParams = getObjectPropertyByPath(spec, 'cookie', openapiSchemas) as any;
-
-    const { mediaType = '', ...bodyParams } = getObjectPropertyByPath(spec, 'body', openapiSchemas) as any;
+    const { mediaType = '', ...bodyParams } = getObjectPropertyByPath(spec, 'body', openapiSchemas) as any || {};
```

Comment:
In line 125, it checks if `bodyParams` exists, like `requestBody: bodyParams && {`.

But if we use default `{}`, `bodyParams` will be an empty object even when `bodyParams` doesn't exist, so it will pass the `bodyParams &&` condition.

Reply:
So I changed this code a little bit https://github.com/ts-spec/tspec/pull/35/commits/0cec67e56383f3c474cdd61f04b663ec6b44e68f

Reply:
With your fix, there is no reason to assign `mediaType` and `bodyParams` separately.  The `mediaType` is a property that should not be included in the generated OAS, and receiving it as a subproperty of `bodyParams` like that results in an invalid OAS.

Check this PR #34